pkg/alert/client.go

```diff
@@ -0,0 +1,343 @@
+package alert
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery/file"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"gopkg.in/yaml.v2"
+
+	"github.com/thanos-io/thanos/pkg/discovery/cache"
+	"github.com/thanos-io/thanos/pkg/discovery/dns"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+const (
+	defaultAlertmanagerPort = 9093
+	alertPushEndpoint       = "/api/v1/alerts"
+	contentTypeJSON         = "application/json"
+)
+
+var userAgent = fmt.Sprintf("Thanos/%s", version.Version)
+
+type AlertingConfig struct {
+	Alertmanagers []AlertmanagerConfig `yaml:"alertmanagers"`
```

Comment:
Just thinking.. It might be nice to also use the `alert_relabel_configs` instead of the `--alert.label-drop` ?
That would provide higher variability to user.
Also adding external labels to the config instead of the `--label` flag?
Not sure how fare we want to take this configuration @bwplotka WDYT?
(Sorry for late comments, I had no time lately)